### PR TITLE
Use the system clipboard only for explicit copy/paste operations. Addresses issue 11907

### PR DESCRIPTION
**Pull request by Tastaturtaste**
# Description

With the introduction of the system clipboard to nushell, many commands changed their behavior from using a local cut buffer to the system clipboard, perhaps surprisingly for many users. (See #11907)
This PR changes most of them back to using the local cut buffer and introduces three commands (`CutSelectionSystem`, `CopySelectionSystem` and `PasteSystem`) to explicitly use the system clipboard.

# User-Facing Changes

Users who in the meantime already used the system clipboard now default back to the local clipboard. To be able to use the system clipboard again they have to append the suffix `system` to their current edit command specified in their keybindings.

# Tests + Formatting

The commands themselves are tested in `reedline`. The changes introduces in nushell are minimal and simply forward from a match on the keybinding name to the command. 
# After Submitting

### CODE REVIEW

**Comment by Tastaturtaste** [comment]
The reedline version is still patched via cargo to the github main branch at this time. I guess the dependency should be changed back to a regular reedline release before this PR gets merged?

**Comment by sholderbach** [comment]
During development we will happily point to the main branch if we want to pull in changes from reedline. We will release a new version when necessary.

File: crates/nu-utils/src/sample_config/default_config.nu [hunk 1]
```diff
@@ -825,18 +825,18 @@ $env.config = {
             event: { edit: capitalizechar }
         }
         {
-            name: copy_selection
+            name: copy_selection_system
```
Comment (sholderbach) [inline crates/nu-utils/src/sample_config/default_config.nu:828]: May be worth adding a comment that this requires the terminal to support the ctrl-shift bindings passing through (and the `system-clipboard` feature, if it is likely to be disabled on a build).

File: Cargo.toml [hunk 2]
```diff
@@ -132,22 +132,22 @@ tempfile = { workspace = true }
...
-  "plugin",
-  "which-support",
-  "trash-support",
-  "sqlite",
-  "mimalloc",
+	"plugin",
+	"which-support",
+	"trash-support",
+	"sqlite",
+	"mimalloc",
```
Comment (sholderbach) [inline Cargo.toml:150]: This should probably be reverted to the formatting we use in the rest of the tomls, also to keep the `git blame` easier to read.

**Review by sholderbach — COMMENTED** [review]
Small nit, but we should be pretty much good to go.

Thanks again!

**Comment by fdncred** [comment]
@Tastaturtaste Do you have time to help with the suggestions made in this PR? We'd like to get this landed asap if possible.

**Comment by fdncred** [comment]
@sholderbach I took a swing at tweaking this PR so we can get it in and have more testing time. Let me know if it looks good to you.

**Review by sholderbach — APPROVED** [review]
Thanks this looks good to me!

**Comment by Tastaturtaste** [comment]
Hey, thanks for pushing this over the finish line. I was pretty swamped with work and other stuff the last couple of days.

**Pull request merged**